src/lib/withMiradorAPI.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+
+import * as actions from '../state/actions';
+import createRootReducer from '../state/reducers';
+
+/**
+ * HoC that initializes a Mirador Application
+ * @param store
+ * @param config
+ * @returns {function(*): {contextType?: React.Context<any>,
+ * new(props: Readonly<P>): HOC, new(props: P, context?: any): HOC, prototype: HOC}}
+ */
+export default function withMiradorAPI(store, config) {
+  // TODO validate config schema before processing / handle invalid or missing configuration
+  // validateConfig(config);
+
+  doInitialization();
+
+  /**
+   * doInitialization
+   */
+  function doInitialization() {
+    processPlugins();
+    dispatchSetConfig(config);
+    fetchManifests();
+    addWindows(config);
+  }
+
+  /**
+   * dispatchSetConfig
+   */
+  function dispatchSetConfig() {
+    const action = actions.setConfig(config);
+    store.dispatch(action);
+  }
+
+  /**
+   * fetchManifests
+   */
+  function fetchManifests() {
+    config.windows.forEach(win => dispatchFetchManifest(win.loadedManifest));
+  }
+
+  /**
+   * dispatchFetchManifest
+   * @param uri
+   */
+  function dispatchFetchManifest(uri) {
+    store.dispatch(actions.fetchManifest(uri));
+  }
+
+  /**
+   * addWindows
+   */
+  function addWindows() {
+    const thumbnailPositions = getThumbnailNavigationPositions(config);
+    thumbnailPositions.forEach((thumbnailNavigationPosition, index) => {
```

Comment:
One thing that was nice about the previous implementation is that we were clearly looping over a list of window configurations to turn them into windows in the workspace; I feel like this refactor has lost some of that clarity.

Reply:
https://github.com/ProjectMirador/mirador/blob/2eee9d41d427d60c3d130f3ee5d1b6ffd1c68666/src/lib/withMiradorAPI.js#L56 will go away when the config validation occurs.  thumbnailNavigation.defaultPosition could be a default value.  convention over configuration is a better pattern.   I suggest fixing https://github.com/ProjectMirador/mirador/blob/2eee9d41d427d60c3d130f3ee5d1b6ffd1c68666/src/lib/withMiradorAPI.js#L77-L87 could add clarity to the whole thing.

there is no error handling or validation in the previous "implementation".  the strategy here is to handle errata and set configuration defaults with schema validation at the API layer.   

FYI: the implementation is where `ReactDOM.render(..)` is.